components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -2054,7 +2056,8 @@ private void sendEmailNotification(String apiId, String apiName, String newVersi
                             .NOTIFICATION_TYPE_NEW_VERSION);
                     new NotificationExecutor().sendAsyncNotifications(notificationDTO);
                 } catch (NotificationException e) {
-                   log.error(e.getMessage());
+                    String msg = "Error occurred while sending Async Notifications";
+                    log.error(msg, e);
```

Comment:
If I remember correctly, we only log at service layer and here we throw the exceptions. Shall we verify that and proceed? @harsha89 ?

Reply:
i think the problem that harsha mention was  If this doesn't use in rest api layer we have to log it in here.